### ISSUE TITLE
Crashes when working directory not writable

### ISSUE THREAD

**Issue by FilipDominec** [Saluev/python-liborigin2#7]
Hi, have you noticed that the library crashes when opening an OPJ file, while the working directory is not allowed to write the temporary `opjfile`?

```
Could not open opjfile.log !
Fatal Python error: Segmentation fault
```

**Comment by Saluev** [comment]
Interesting. Have you tried to comment out all `print` statements in `liborigin.pyx`?

**Comment by FilipDominec** [comment]
Not yet - should it help prevent segfaults in non-writable directories? :)

**Comment by Saluev** [comment]
Actually, no. But that was my first guess.
I found it out: you need to compile the C++ code with `NO_CODE_GENERATION_FOR_LOG` defined.

**Comment by FilipDominec** [comment]
Thank you, this is obviously the cause of troubles - if opjfile.log can not be open, the routine returns without initializing necessary variables, which later cause a program crash. Do you think the output to `opjfile.log` could be just skipped and the program could proceed further?

**Comment by Saluev** [comment]
I tell you -- add a `-DNO_CODE_GENERATION_FOR_LOG` flag to C++ compiler and the output to log file will not be compiled.

**Comment by FilipDominec** [comment]
I agree the problem can be circumvented through re-compiling whole library, but I believe it is a good design practice to make a library robust enough to work in most usual cases, including insufficient writing permissions.

**Comment by Saluev** [comment]
Yes. I meant, you could add that flag to default build config and create a pull request.

**Comment by FilipDominec** [comment]
Reading the source I can see that the file is written to at multiple places, so just changing the flag as you suggest seems the best solution for now. There are more urgent things that need fixing.
